src/lib/serviceRoutinesV2/getEntities.cpp

```diff
@@ -61,17 +60,58 @@ std::string getEntities
   ParseData*                 parseDataP
 )
 {
-  std::string  answer;
-  Entities     entities;
+  using namespace std;
+  typedef vector<string>  vstr;
+  typedef vstr::size_type vstr_sz;
+
+  Entities entities;
+  string   answer;
+
+  // optional parameter for list of IDs
+  string paramIDs     = ciP->uriParam["id"];
+  string paramPattern = ciP->uriParam["idPattern"];
+  string pattern      = ".*"; // all entities, default value
 
+  if (paramPattern != "" && paramIDs != "")
+  {
+    ciP->httpStatusCode = SccBadRequest;
+    answer = "{\"error\":\"Incompatible parameters: id, IdPattern\"}";
```

Comment:
OrionError is actually prepared for v2, so ... I would prepose to use it ...

    std::string OrionError::render(ConnectionInfo* ciP, const std::string& _indent)
    {
      //
      // For API version 2 this is pretty easy ...
      //
      if (ciP->apiVersion == "v2")
      {
        if ((ciP->httpStatusCode == SccOk) || (ciP->httpStatusCode == SccNone))
        {
          ciP->httpStatusCode = SccBadRequest;
        }

        reasonPhrase = errorStringForV2(reasonPhrase);
        return "{" + JSON_STR("error") + ":" + JSON_STR(reasonPhrase) + "," + JSON_STR("description") + ":" + JSON_STR(details) + "}";
      }

Is that what you refer to?